packages/near-api-js/src/wallet-account.ts

```diff
@@ -102,11 +102,21 @@ export class WalletConnection {
             });
         }
         this._near = near;
+        const DEPRECATED_authDataKey = appKeyPrefix + LOCAL_STORAGE_KEY_SUFFIX;
+        const DEPRECATED_authData = JSON.parse(window.localStorage.getItem(DEPRECATED_authDataKey));
+
+        appKeyPrefix = appKeyPrefix || near.config.contractName || 'default';
         const authDataKey = appKeyPrefix + LOCAL_STORAGE_KEY_SUFFIX;
+
+        // Migrate dApps that previously used a faulty authDataKey.
+        if (DEPRECATED_authData && DEPRECATED_authDataKey != authDataKey) {
+            window.localStorage.removeItem(DEPRECATED_authDataKey);
+            window.localStorage.setItem(authDataKey, JSON.stringify(DEPRECATED_authData));
+        }
+
```

Comment:
Just to make this logic easier to follow:

```suggestion
        const DEPRECATED_appKeyPrefix = appKeyPrefix;
        const finalAppKeyPrefix = appKeyPrefix || near.config.contractName || 'default';

        const DEPRECATED_authDataKey = DEPRECATED_appKeyPrefix + LOCAL_STORAGE_KEY_SUFFIX;
        const authDataKey = finalAppKeyPrefix + LOCAL_STORAGE_KEY_SUFFIX;

        if (DEPRECATED_authDataKey != authDataKey) {
            const DEPRECATED_authData = JSON.parse(window.localStorage.getItem(DEPRECATED_authDataKey));
            if (DEPRECATED_authData) {
                window.localStorage.removeItem(DEPRECATED_authDataKey);
                window.localStorage.setItem(authDataKey, JSON.stringify(DEPRECATED_authData));
            }
        }
```